### ISSUE TITLE
Implement Money and Currency SPI for Bitcoin

### ISSUE THREAD

**Issue by msgilligan** [JavaMoney/javamoney-shelter#4]
Is there an existing project to do this? If not, can we start one here?

**Comment by msgilligan** [comment]
Is see there is example code for a `BitcoinProvider` in the [JavaMoney User Guide]( https://github.com/JavaMoney/jsr354-ri/blob/master/src/main/asciidoc/userguide.adoc#214-registering-additional-currency-units) Does that code live in a repository anywhere?

**Comment by keilw** [comment]
I don't think so, since the User Guide provides examples of how it may be used. Some parts of "Shelter" especially those related to defunct providers like Mt.Gox were initially contributed by JUG Chennai, mainly Raj who since also joined the EG on behalf of his JUG. I improved those after adding them to the Shelter e.g. some hard-coded references to bypass proxy, etc. that made no sense. However, as these used an outdated provider the API and RI evolved a bit at the same time, so some of these practices probably should live in the Shelter but nobody found time to "adopt" them there so far.
You'd be more than welcome to suggest a concrete SPI implementation there.

**Comment by msgilligan** [comment]
PR #7 is my attempt at a port of the BitcoinProvider from the User's Guide to the current API/SPI. I wrote some tests and they do pass, but I'm new to the SPI and not very confident I did everything correctly.

**Comment by msgilligan** [comment]
Playing around with "Assignee" setting. It seems only one assignee is allowed. I'm going to contribute to this one but am definitely looking for help and guidance from the Moneta designer/developers.
